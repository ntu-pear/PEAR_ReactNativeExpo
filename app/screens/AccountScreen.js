import React, { useContext } from 'react';
import { Text, View } from 'react-native';
import AuthContext from 'app/auth/context';
import AppButton from 'app/components/AppButton';
import authStorage from 'app/auth/authStorage';

function AccountScreen() {
  const { user, setUser } = useContext(AuthContext);

  const onPressLogOut = async () => {
    setUser(null);
    await authStorage.removeToken();
  };

  return (
    <View>
      <Text>This is my accountscreen</Text>
      <Text>{user.sub}</Text>
      <AppButton title="Logout" color="red" onPress={onPressLogOut} />
    </View>
  );
}

export default AccountScreen;
