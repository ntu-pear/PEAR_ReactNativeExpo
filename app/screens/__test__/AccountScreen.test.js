/**
 * @jest-environment node
 */
import AuthContext from 'app/auth/context';
import AccountScreen from 'app/screens/AccountScreen';
import {
  cleanup,
  fireEvent,
  render,
  screen,
} from '@testing-library/react-native';

const renderAccountScreen = () => {
  return render(
    <AuthContext.Provider
      value={{
        user: {
          sub: 'Jessica Sim',
        },
      }}
    >
      <AccountScreen />
    </AuthContext.Provider>,
  );
};

afterEach(cleanup);
describe('Testing Account Screen', () => {
  it('Account screen snapshot should match', () => {
    expect(renderAccountScreen().toJSON()).toMatchSnapshot();
  });
  it('Logout screen snapshot should match', () => {
    const accountScreen = renderAccountScreen();
    fireEvent.press(screen.getByText('Logout'));
    expect(accountScreen.toJSON()).toMatchSnapshot();
  });
});
