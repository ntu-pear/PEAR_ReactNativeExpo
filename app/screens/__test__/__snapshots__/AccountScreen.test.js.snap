// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Account Screen Account screen snapshot should match 1`] = `
<View>
  <Text>
    This is my accountscreen
  </Text>
  <Text>
    Jessica Sim
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FD4734",
        "borderRadius": 25,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 15,
        "widht": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#F3ECE4",
          "fontSize": 18,
          "fontWeight": "bold",
          "textTransform": "uppercase",
        }
      }
    >
      Logout
    </Text>
  </View>
</View>
`;

exports[`Testing Account Screen Logout screen snapshot should match 1`] = `
<View>
  <Text>
    This is my accountscreen
  </Text>
  <Text>
    Jessica Sim
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FD4734",
        "borderRadius": 25,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 15,
        "widht": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#F3ECE4",
          "fontSize": 18,
          "fontWeight": "bold",
          "textTransform": "uppercase",
        }
      }
    >
      Logout
    </Text>
  </View>
</View>
`;
